tests/neg/i13820.scala

```diff
@@ -0,0 +1,5 @@
+trait Expr { type T }
+
+def foo[A](e: Expr { type T = A }) = e match
+  case e1: Expr { type T <: Int } =>
+    val i: Int = ??? : e1.T // error: unrealizable bounds
```

Comment:
I had a look at this one and it's because its bounds are `= A` and `<: Int`, i.e. `>: A <: A & Int` and it doesn't believe `A <: Int`.  But I think given the match that should be recorded as a fact, from matching the pattern.

The case is totally uncheckable of course, but maybe the compiler should infer that `A <: Int` and not emit this error.

I thought this, i.e. refined type patterns, would be a component @Linyxus' #14754 work but, surprisingly to me, they're not. 🤔

Reply:
> I thought this, i.e. refined type patterns, would be a component @Linyxus' #14754 work but, surprisingly to me, they're not. 🤔


Yes, ideally that should be done in #14754. But this is currently a missing part of it because we have not figured out a proper way to get the bound symbol in the pattern (if exists). The tricky part is that the symbol of the pattern only gets created *after* we perform GADT reasoning. I have some ideas of how to do this, maybe we could discuss them when reviewing #14754.